mage_ai/io/config.py

```diff
@@ -0,0 +1,229 @@
+
+        Returns:
+            Union(bytes, str): The secret stored under `secret_id` in AWS secret manager. If secret is:
+            - a binary value, returns a `bytes` object
+            - a string value, returns a `string` object
+        """
+        response = self.__get_secret(secret_id, version_id, version_stage_label)
+        if 'SecretBinary' in response:
+            return response['SecretBinary']
+        else:
+            return response['SecretString']
+
+    def __get_secret(
+        self, secret_id: Union[ConfigKey, str], version_id=None, version_stage_label=None
+    ) -> Dict:
+        """
+        Get secret with ID `secret_id`. Can also specify the version of the secret to get.
+        If
+        - both `version_id` and `version_stage_label` are specified, both must agree on the
+          secret version
+        - neither of `version_id` or `version_stage_label` are specified, a check is made for
+          the current version
+        - one of `version_id` and `version_stage_label` are specified, the associated version
+          is loaded
+
+        Args:
+            secret_id (str): ID of the secret to load
+            version_id (str, Optional): ID of the version of the secret to load. Defaults to None.
+            version_stage_label (str, Optional): Staging label of the version of the secret to load.
+            Defaults to None.
+
+        Returns:
+            Dict: response object returned by AWS Secrets Manager API
+        """
+        try:
+            return self.client.get_secret_value(
+                SecretID=secret_id,
+                VersionId=version_id,
+                VersionStage=version_stage_label,
+            )
+        except ClientError as error:
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
+                return None
+            raise RuntimeError(f'Error loading config: {error.response["Error"]["Message"]}')
+
+
+class EnvironmentVariableLoader(BaseConfigLoader):
+    def contains(self, env_var: Union[ConfigKey, str]) -> bool:
+        """
+        Checks if the environment variable is defined.
+        Args:
+            key (Union[ConfigKey, str]): Name of the configuration setting to check existence of.
+
+        Returns:
+            bool: Returns true if configuration setting exists, otherwise returns false.
+        """
+        return env_var in os.environ
+
+    def get(self, env_var: Union[ConfigKey, str]) -> Any:
+        """
+        Loads the config setting stored under the environment variable
+        `env_var`.
+
+        Args:
+            env_var (str): Name of the environment variable to load configuration setting from
+
+        Returns:
+            Any: The configuration setting stored under `env_var`
+        """
+        return os.getenv(env_var)
+
+
+class ConfigFileLoader(BaseConfigLoader):
+    def __init__(
+        self, filepath: os.PathLike = './default_repo/io_config.yaml', profile='default'
```

Comment:
do not use default_repo here. default_repo is only for development. use `get_repo_path()` instead.

Reply:
Fixed.